pkg/controller/petset/pet_set.go

```diff
@@ -343,6 +343,11 @@ func (psc *PetSetController) syncPetSet(ps *apps.PetSet, pets []*api.Pod) (int,
 			if err == nil {
 				numPets++
 			}
+			switch err.(type) {
+			case *unhealthyPetBlock:
```

Comment:
confused by your comment, what am i missing in this example? https://play.golang.org/p/4nswvOvc0m

I think you can just not return the pointer and not define the method with a pointer interface and it should work.